text_extensions_for_pandas/array/char_span.py

```diff
@@ -73,9 +73,19 @@ def __repr__(self) -> str:
                                        textwrap.shorten(self.covered_text, 80))
 
     def __eq__(self, other):
-        return (self.begin == other.begin
-                and self.end == other.end
-                and self.target_text == other.target_text)
+        if isinstance(other, CharSpan):
+            return (self.begin == other.begin
+                    and self.end == other.end
+                    and self.target_text == other.target_text)
+        elif isinstance(other, CharSpanArray):
+            return other == self
+        else:
+            # Different type ==> not equal
+            return False
+
+    def __hash__(self):
+        result = hash((self.target_text, self.begin, self.end))
```

Comment:
would it make sense to cache a hash of the target text and use that whenever computing this hash, and also for equality checking?

Reply:
Yes, adding caching to the hash function would make sense. Created #27 to cover this.